demos/http/http_demo_basic_tls/http_demo_basic_tls.c

```diff
@@ -0,0 +1,810 @@
+                            const char * pPath )
+{
+    /* Return value of this method. */
+    int returnStatus = EXIT_SUCCESS;
+
+    /* Configurations of the initial request headers that are passed to
+     * #HTTPClient_InitializeRequestHeaders. */
+    HTTPRequestInfo_t requestInfo;
+    /* Represents a response returned from an HTTP server. */
+    HTTPResponse_t response;
+    /* Represents header data that will be sent in an HTTP request. */
+    HTTPRequestHeaders_t requestHeaders;
+
+    /* Return value of all methods from the HTTP Client library API. */
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    /* Initialize all HTTP Client library API structs to 0. */
+    memset( &requestInfo, 0, sizeof( requestInfo ) );
+    memset( &response, 0, sizeof( response ) );
+    memset( &requestHeaders, 0, sizeof( requestHeaders ) );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = strlen( pHost );
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = strlen( pMethod );
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                      &requestInfo );
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Initialize the response object. The same buffer used for storing
+         * request headers is reused here. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        LogInfo( ( "Sending HTTP %s request to %s%s...",
+                   pMethod, SERVER_HOST, pPath ) );
+        LogDebug( ( "Request Headers:\n%.*s",
+                    ( int32_t ) requestHeaders.headersLen,
+                    ( char * ) requestHeaders.pBuffer ) );
+        LogDebug( ( "Request Body:\n%.*s\n",
+                    ( int32_t ) REQUEST_BODY_LENGTH,
+                    REQUEST_BODY ) );
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( pTransportInterface,
+                                      &requestHeaders,
+                                      ( uint8_t * ) REQUEST_BODY,
+                                      REQUEST_BODY_LENGTH,
+                                      &response,
+                                      0 );
+    }
+    else
+    {
+        LogError( ( "Failed to initialize HTTP request headers: Error=%s.",
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
```

Comment:
If all are logs, can we not combine? 

  LogInfo( ( "Received HTTP response from %s/%s. Response Headers:\n%.*s. Response Status:\n%u. Response Body:\n%.*s\n" ) )